samples/template-operator/README.md

```diff
@@ -36,117 +36,206 @@ MODULE_NAME ?= template
    On top, `.status` object could contain other relevant properties as per your requirements.
-2. Next, check method `Reconcile()` inside the [SampleController](./controllers/sample_controller.go), which demonstrates how `.status.state` properties could be set, depending upon your logic.
-3. The `.status.state` value has a literal sense behind it, so use them appropriately.
+2. The `.status.state` values have literal meaning behind them, so use them appropriately.
 
-## Grafana dashboard
+### Grafana dashboard
 
 By the following command, two grafana dashboard files with controller related metrics will be generated under `/operator/grafana` folder.
 
-```
+```shell
 make grafana-dashboard
 ```
 
 For how to import dashboard, please read [official grafana guide](https://grafana.com/docs/grafana/latest/dashboards/export-import/#import-dashboard).
 This feature is supported by [kubebuilder grafana plugin](https://book.kubebuilder.io/plugins/grafana-v1-alpha.html).
+
+## Abstraction: Reconciliation and Status handling
+
+For simple use cases where a `module operator` should install a `module helm chart(s)` and set the state of the corresponding `module CR` accordingly, a declarative approach is useful.
+This approach will enable orchestration of Kubernetes resources so that module owners can concentrate on their specific logic.
+
+### Steps:
+
+1. Refer to [API definition](./operator/api/v1alpha1/sample_types.go) of `SampleCR` and implement `Status` sub-resource similarly.
+   This `Status` type definition is sourced from the `module-manager` declarative library and contains all valid `.status.state` values as discussed in the previous sections.
+   ```yaml
+    Status types.Status `json:"status,omitempty"`
+   ```
+   
+2. Ensure the module CR implements the `module-manager` declarative library's resource requirements, represented by `types.CustomObject`. Also implement missing interface methods.
+   ```go
+   var _ types.CustomObject = &Sample{}
+
+   func (s *Sample) GetStatus() types.Status {
+        return s.Status
+   }
+   
+   func (s *Sample) SetStatus(status types.Status) {
+        s.Status = status
+   }
+   
+   func (s *Sample) ComponentName() string {
+        return "sample-component-name"
+   }
+   ```
+
+3. Refer to the [controller implementation](./operator/controllers/sample_controller.go). 
+Instead of implementing the default reconciler interface, as provided by `kuberbuilder`, include the `module-manager` declarative reconciler.
+   ```go
+   // SampleReconciler reconciles a Sample object
+   type SampleReconciler struct {
+        declarative.ManifestReconciler
+        client.Client
+        Scheme *runtime.Scheme
+        *rest.Config
+   }
+   ```
+   Notice there is not `Reconcile()` method implemented in this controller, since the logic is abstracted within the declarative reconciler.
```

Comment:
```suggestion
   Notice there is no `Reconcile()` method implemented in this controller, since the logic is abstracted within the declarative reconciler.
```